dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetSearcher.java

```diff
@@ -0,0 +1,140 @@
+package com.dotmarketing.portlets.fileassets.business;
+
+import javax.annotation.Nonnull;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.liferay.portal.model.User;
+import io.vavr.control.Try;
+
+
+@JsonDeserialize(builder = FileAssetSearcher.Builder.class)
+public class FileAssetSearcher {
+
+    final Host host;
+    final Folder folder;
+    final long language;
+    final int offset, limit;
+    final String searchTerm;
+    final User user;
+    final boolean respectFrontendRoles;
+    final boolean live;
+    private FileAssetSearcher(Builder builder) {
+
+        
+        this.host = builder.host ;
+        this.folder = builder.folder;
+        this.language = builder.language;
+        this.offset = builder.offset;
+        this.limit = builder.limit;
+        this.searchTerm = builder.searchTerm;
+        this.user = builder.user;
+        this.respectFrontendRoles = builder.respectFrontendRoles;
+        this.live=builder.live;
+    }
+    /**
+     * Creates builder to build {@link FileAssetSearcher}.
+     * @return created builder
+     */
+    
+    public static Builder builder() {
+        return new Builder();
+    }
+    /**
+     * Creates a builder to build {@link FileAssetSearcher} and initialize it with the given object.
+     * @param fileAssetSearcher to initialize the builder with
+     * @return created builder
+     */
+    
+    public static Builder from(FileAssetSearcher fileAssetSearcher) {
+        return new Builder(fileAssetSearcher);
+    }
+    /**
+     * Builder to build {@link FileAssetSearcher}.
+     */
+    
+    public static final class Builder {
+        private Host host;
+        private Folder folder;
+        private long language;
+        private int offset;
+        private int limit;
+        private String searchTerm;
+        private User user;
+        private boolean respectFrontendRoles;
+        private boolean live=false;
```

Comment:
Codacy found an issue: [Avoid using redundant field initializer for 'live'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6325390)